benchmark/scenarios/write_request_load.py

```diff
@@ -0,0 +1,349 @@
+        rate to be reached.
+    """
+
+    def __init__(
+        self, reactor, cluster, request_rate=10,
+        sample_size=DEFAULT_SAMPLE_SIZE, timeout=45
+    ):
+        self._maintained = Deferred()
+        self.reactor = reactor
+        self.control_service = cluster.get_control_service(reactor)
+        self.request_rate = request_rate
+        self.timeout = timeout
+        self.rate_measurer = WRateMeasurer(sample_size)
+        self.max_outstanding = 10 * request_rate
+        self._dataset_id = ""
+        # Send requests per second
+        self.loop = LoopingCall.withCount(self._request_and_measure)
+        self.loop.clock = self.reactor
+        # Once the expected rate is reached, we will start monitoring
+        # the scenario inside this loop
+        self.monitor_loop = LoopingCall(self.check_rate)
+        self.monitor_loop.clock = self.reactor
+
+    def start(self):
+        """
+        Executes the setup and starts running the write scenario.
+
+        :return: A Deferred that fires when the desired scenario is
+            established (e.g. that a certain load is being applied).
+        """
+        # List all the nodes registered in the control service
+        d = self.control_service.list_nodes()
+        d.addCallback(self._get_dataset_node)
+        # Once we have the list of nodes, we will create a dataset.
+        # We cannot start the scenario until we have a working dataset, so
+        # `_create_dataset` will work like a setup of the write scenario
+        d.addCallback(self._create_dataset)
+        # Once we have setup all we need, we can start running the scenario
+        d.addCallback(self.run_scenario)
+        return d
+
+    def _create_dataset(self, node):
+        """
+        Creates a dataset in the node given.
+
+        :param node: node where we want the dataset.
+
+        :return: A Deferred that fires when the dataset has been created.
+
+        :raises: `WDataseCreationTimeout` if the creation goes wrong.
+        """
+        self.dataset_node = node
+        creating = self.control_service.create_dataset(
+            primary=node.uuid)
+
+        # Not sure about handling errors and timeout in the same errback.
+        # How could I handle them differently?
+        def handle_timeout_and_errors(failure):
+            failure.trap(CancelledError)
+            raise WDataseCreationTimeout()
+
+        timeout(self.reactor, creating, self.timeout)
+
+        creating.addErrback(handle_timeout_and_errors)
+        return creating
+
+    def _get_dataset_node(self, nodes):
+        """
+        Selects the node where the dataset will be created.
+
+        :param nodes: list of `Node` where we will chose one
+            to create the dataset.
+
+        :return: the selected `Node`.
+        :raise: `WNoNodesFound` if the given list of nodes was empty.
+        """
+        if not nodes:
+            raise WNoNodesFound()
+        return nodes[0]
```

Comment:
Use `random.choice` to select the node. See https://github.com/ClusterHQ/flocker/blob/9ea105e7552c2f358a4feb5af4e39323e49c6085/benchmark/operations/create_dataset.py#L50-L55